engine/overworld/map_objects.asm

```diff
@@ -6,11 +6,14 @@ DeleteMapObject::
 	add hl, bc
 	ld a, [hl]
 	push af
+	push hl
```

Comment:
I think instead of `push hl / ... / pop hl / ld [hl], -1`, you can do `ld [hl], -1 / inc hl` and change `ld bc, OBJECT_LENGTH` to `ld bc, OBJECT_LENGTH - 1`.

Reply:
We couldn't do exactly that.. but I was able to optimize it some I think.